### PR TITLE
main menu and diagram menu implemented with tab autocomplete

### PR DESCRIPTION
**Pull request by lillyfetzer**

### CODE REVIEW

File: src/Controller/MenuController.java [hunk 1]
```diff
@@ -335,4 +344,44 @@ public static void editRelationshipsControl(boolean shouldTerminate, final Diagr
             }
         }
     }
+    public static boolean autoCompleteOptionsControl(final String command, final Diagram diagram, boolean shouldTerminate) {
```
Comment (Ancross333) [inline src/Controller/MenuController.java:347]: Autocomplete options control is a misleading name, as it doesn't seem to be related to autocomplete as much as it is just executing the command

File: src/application/CLI/AutoComplete.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+public class AutoComplete {
+
+    private Terminal terminal;
+    private LineReader lineReader;
+
+    public AutoComplete() {
+        try {
+            this.terminal = TerminalBuilder.builder()
+                            .build();
+            /*this.lineReader = LineReaderBuilder.builder()
```
Comment (Ancross333) [inline src/application/CLI/AutoComplete.java:20]: Is there a reason this commented code is left behind?

File: src/application/CLI/AutoComplete.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    public String getCommands(){
+        String line = "";
+        try {
+
+            // Main loop to read input
+            while (true) {
+                //String prompt = "Type Command (press Tab for autocomplete): ";
+                //String prompt2 = "Type what entity to edit (press Tab for autocomplete): ";
+                line = this.lineReader.readLine();
+                //String line2 = lineReader2.readLine(prompt2);
```
Comment (Ancross333) [inline src/application/CLI/AutoComplete.java:95]: More commented code

File: src/application/CLI/CommandLineInterface.java [hunk 4]
```diff
@@ -33,9 +35,18 @@ public void launchUmlEditor(){
...
         boolean shouldTerminate = false;
         while (!shouldTerminate) {
-            int userChoice = getUserChoice();
+            String userChoice = getUserChoice();
+            if(userChoice.isEmpty()){
+                continue;
+            }
+            if (!Character.isDigit(userChoice.charAt(0))){
+                //autocomplete methods
+                shouldTerminate = AutoCompleteControl(userChoice.trim(), shouldTerminate);
```
Comment (Ancross333) [inline src/application/CLI/CommandLineInterface.java:44]: If you're assigning shouldTerminate to the value that AutoCompleteControl is returning (true in the case of exit being entered), there's no reason to pass it down and modify it in the method

File: src/application/CLI/CommandLineInterface.java [hunk 5]
```diff
@@ -454,4 +500,37 @@ public static void editClassMenuHelp(){
             Option 6 - Return to Diagram Menu: returns the user to the diagram menu holding the class
                 """);
     }
+
+    public static boolean AutoCompleteControl(final String command, boolean shouldTerminate) {
```
Comment (Ancross333) [inline src/application/CLI/CommandLineInterface.java:504]: Same naming issue as the other method. Since this isn't related to auto-complete the method name is misleading

**Review by Ancross333 — CHANGES_REQUESTED** [review]
Just some minor issues

File: src/Controller/MenuController.java (hunk 1 above)
Comment (lillyfetzer) [reply src/Controller/MenuController.java:347]: True, I think I named it that initially before really thinking what it did but that's a good point.

File: src/application/CLI/AutoComplete.java (hunk 2 above)
Comment (lillyfetzer) [reply src/application/CLI/AutoComplete.java:20]: Nope I must have forgot to delete it.

File: src/application/CLI/CommandLineInterface.java (hunk 4 above)
Comment (lillyfetzer) [reply src/application/CLI/CommandLineInterface.java:44]: Good point, I will remove the boolean passed in.

File: src/Controller/MenuController.java [hunk 6]
```diff
@@ -364,7 +364,7 @@ public static boolean autoCompleteOptionsControl(final String command, final Dia
                 editClassSubMenu(false, currentClass, diagram);
                 break;
             case("edit-relationships"):
-                editRelationshipsControl(shouldTerminate, diagram);
+                editRelationshipsControl(false, diagram);
```
Comment (Ancross333) [inline src/Controller/MenuController.java:367]: If false is going to be passed in here everytime, is there any reason the boolean parameter needs to exist?

**Review by Ancross333 — CHANGES_REQUESTED** [review]
One more issue

**Review by Ancross333 — APPROVED** [review]
Looks good
